### DISCUSSION TITLE
Rule proposal: `template/no-restricted-syntax`

### DISCUSSION THREAD

**Discussion by rafaelss95** [angular-eslint/angular-eslint#831] [New Rule Proposals]
Similar to https://eslint.org/docs/rules/no-restricted-syntax, but for templates. Taking https://github.com/angular-eslint/angular-eslint/issues/308 as an example:

```json
{
  "rules": {
    "@angular-eslint/template/no-restricted-syntax": [
      "error",
      {
        "selector": "Element[name='button']:not(:has(:matches(BoundAttribute, TextAttribute)[name='type']))",
        "message": "Missing an explicit type attribute for button"
      },
      {
        "selector": "Element[name='button'] :matches(BoundAttribute, TextAttribute)[name='type'][value!=/^(button|reset|submit)$/]",
        "message": "Invalid value for button type attribute"
      },
    ]
  }
}
```

**Comment by JamesHenry** [comment]
Nice idea, @rafaelss95! I think it would be a good addition

**Reply by neodescis** [reply]
@JamesHenry, why did the original issue get closed as completed when the PR was never accepted? It's confusing Gemini to no end 😆

**Reply by JamesHenry** [reply]
<img width="361" height="96" alt="image" src="https://github.com/user-attachments/assets/08943ecd-2c4a-4696-b67d-2481840d1bf1" />

**Reply by neodescis** [reply]
Yes, obviously I saw that, as I made it here to the discussion. I guess my point is, should the issue be reopened given that the PR was abandoned?

**Reply by JamesHenry** [reply]
GitHub closes the issue when it gets migrated to a discussion. Discussions are where rule proposals belong. If you want to work on it, go ahead. The PRs Welcome label has been on this for 5 years.
